code/modules/projectiles/projectile.dm

```diff
@@ -123,23 +137,98 @@
 	return 1 //Bullets don't drift in space
 
 
-/obj/item/projectile/proc/fire()
-	spawn()
-		while(loc)
-			if(kill_count < 1)
-				qdel(src)
-				return
-			if(!paused)
+/obj/item/projectile/proc/fire(var/setAngle)
+	if(setAngle) Angle = setAngle
+	if(!legacy)
+		spawn() //New projectile system
+			while(loc)
+				if(kill_count < 1)
+					qdel(src)
+					return
 				kill_count--
 				if((!( current ) || loc == current))
 					current = locate(Clamp(x+xo,1,world.maxx),Clamp(y+yo,1,world.maxy),z)
-				step_towards(src, current)
-				if((original && original.layer>=2.75) || ismob(original))
+
+				if(!Angle)
+					Angle=round(Get_Angle(src,current))
+				// world << "[Angle] angle"
+				// overlays.Cut()
+				// var/icon/I=new(initial(icon),icon_state) //using initial(icon) makes sure that the angle for that is reset as well
+				// I.Turn(Angle)
+				// I.DrawBox(rgb(255,0,0,50),1,1,32,32)
+				// icon = I
+				if(spread) //Chaotic spread
+					Angle += (rand() - 0.5) * spread
+				var/matrix/M = new//matrix(transform)
+				M.Turn(Angle)
+				transform = M
+
+				var/Pixel_x=round(sin(Angle)+16*sin(Angle)*2)

```

Comment:
I would say top of proc if you only use the variables here and to of page if you use them throughout the page.